CMakeLists.txt

```diff
@@ -31,20 +32,25 @@ add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third-party/re2)
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third-party/sentencepiece)
 
 # Configure PCRE2
-set(PCRE2_BUILD_PCRE2_8 ON)
-set(PCRE2_BUILD_PCRE2_16 OFF)
-set(PCRE2_BUILD_PCRE2_32 OFF)
-set(PCRE2_BUILD_TESTS OFF)
-set(PCRE2_BUILD_PCRE2GREP OFF)
-set(PCRE2_BUILD_PCRE2TEST OFF)
-set(PCRE2_BUILD_PCRE2GPERF OFF)
-set(PCRE2_BUILD_DOCS OFF)
-set(PCRE2_BUILD_LIBPCRE2_PDB OFF)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third-party/pcre2)
+if(SUPPORT_REGEX_LOOKAHEAD)
+    set(PCRE2_BUILD_PCRE2_8 ON)
+    set(PCRE2_BUILD_PCRE2_16 OFF)
+    set(PCRE2_BUILD_PCRE2_32 OFF)
+    set(PCRE2_BUILD_TESTS OFF)
+    set(PCRE2_BUILD_PCRE2GREP OFF)
+    set(PCRE2_BUILD_PCRE2TEST OFF)
+    set(PCRE2_BUILD_PCRE2GPERF OFF)
+    set(PCRE2_BUILD_DOCS OFF)
+    set(PCRE2_BUILD_LIBPCRE2_PDB OFF)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/third-party/pcre2)
+endif()
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ${_pic_flag})
 
 file(GLOB tokenizers_source_files ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp)
+if(NOT SUPPORT_REGEX_LOOKAHEAD)
+    list(REMOVE_ITEM tokenizers_source_files ${CMAKE_CURRENT_SOURCE_DIR}/src/pcre2_regex.cpp)
+endif()
```

Comment:
Can you add instead of remove

Reply:
It's time to explicitly list out the source files, instead of simply GLOB

Reply:
So instead of the glob above this line, specify each individual .cpp?

Reply:
Beat me to it. Sounds good